### ISSUE TITLE
Hang-up with footnote/pagefloat and ::after pseudo element

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#869]
Vivliostyle.js hangs up with footnote and ::after pseudo element.

Test case: https://gist.githubusercontent.com/MurakamiShinyu/5451789a627730e5afb829edb7ad87bf/raw/footnote-pseudo-bug1.html

```css
    .footnote {
      float: footnote;
    }
    p::after {
      content: "!";
    }
```

```html
  <p>TEST<span class="footnote">Footnote</span></p>
```

Test with Vivliostyle Viewer (v2.14.1): https://vivliostyle.github.io/viewer/v2.14.1/#src=https://gist.githubusercontent.com/MurakamiShinyu/5451789a627730e5afb829edb7ad87bf/raw/footnote-pseudo-bug1.html

**Comment by MurakamiShinyu** [comment]
The same problem occurs with page floats:

Test case:
https://gist.githubusercontent.com/MurakamiShinyu/45081a2e9a572927c7d17568e647999b/raw/pagefloat-pseudo-bug1.html

Test with Vivliostyle Viewer (v2.14.1): https://vivliostyle.github.io/viewer/v2.14.1/#src=https://gist.githubusercontent.com/MurakamiShinyu/45081a2e9a572927c7d17568e647999b/raw/pagefloat-pseudo-bug1.html

**Comment by MurakamiShinyu** [comment]
Fixed in #872 

Test:

- https://vivliostyle-git-fix-issue869-vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/5451789a627730e5afb829edb7ad87bf/raw/footnote-pseudo-bug1.html
- https://vivliostyle-git-fix-issue869-vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/45081a2e9a572927c7d17568e647999b/raw/pagefloat-pseudo-bug1.html
